go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -343,7 +344,68 @@ func (m *MasterController) setupHybridLRPolicySharedGw(nodeSubnets []*net.IPNet,
 			Nexthops: []string{drIP.String()},
 			Match:    matchStr,
 		}
-		logicalRouterPolicyRes := []nbdb.LogicalRouterPolicy{}
+		// logicalRouterPolicyRes := []nbdb.LogicalRouterPolicy{}
+
+		// Logic route policy to steer packet from external to nodePort service backed by non-ovnkube pods to hybrid overlay nodes
+		logicalPort := ovntypes.RouterToSwitchPrefix + nodeName
+		if err := util.CreateMACBinding(m.sbClient, logicalPort, ovntypes.OVNClusterRouter, portMac, drIP); err != nil {
+			return fmt.Errorf("failed to create MAC Binding for hybrid overlay: %v", err)
+		}
+		gwLRPIfAddrs, err := util.GetLRPAddrs(m.nbClient, ovntypes.GWRouterToJoinSwitchPrefix+ovntypes.GWRouterPrefix+nodeName)
+		if err != nil {
+			return err
+		}
+		gwLRPIfAddr, err := util.MatchIPNetFamily(utilnet.IsIPv6CIDR(hybridCIDR), gwLRPIfAddrs)
+		if err != nil {
+			return err
+		}
+		grMatchStr := fmt.Sprintf(`%s.src == %s && %s.dst == %s`,
+			L3Prefix, gwLRPIfAddr.String(), L3Prefix, hybridCIDR)
+		grLogicalRouterPolicy := nbdb.LogicalRouterPolicy{
+			Priority: intPriority,
+			ExternalIDs: map[string]string{
+				"name": ovntypes.HybridSubnetPrefix + nodeName + "-gr",
+			},
+			Action:   nbdb.LogicalRouterPolicyActionReroute,
+			Nexthops: []string{drIP.String()},
+			Match:    grMatchStr,
+		}
+
+		// grLogicalRouterPolicyRes := []nbdb.LogicalRouterPolicy{}
+
+		intLRPIfAddr := util.GetNodeHybridOverlayIfAddr(nodeSubnet)
+		if err != nil {
+			return fmt.Errorf("failed to add a static route in cluster router with distributed "+
+				"router as the nexthop: %v", err)
+		}
+
+		// Static route to steer packets from external to nodePort service backed by pods on hybrid overlady node.
+		// This route is to used for triggering above route policy
```

Comment:
why do you need a static route exactly? This comment doesn't really make sense. logical route policy is always evaluated after logical routes.

Reply:
Without the static route, the packet cannot hit any items in the routing table, and it will be dropped and will not be evaluated by the logical route policy. So the static route is to ensure the packet can be matched in the routing table.